### DISCUSSION TITLE
User defined or Spatially-Dependent Temparature Profiles

### DISCUSSION THREAD

**Discussion by appheadspace** [firemodels/fds#13704] [Q&A]
Hi, I'm studying natural convection in 2D cavity with a isothermal vertical walls and insulated horizontal walls. Due to some practical reasons, horizontal walls are not perfectly insulated. So, temp values are measured during experiment and given at some spatial intervals. I've tried "11.3 Spatially-Dependent Velocity Profiles" approach with RAMP_T, but it is not working (showing temp =0 for horizontal walls instead of given temp distribution. Kindly help !

I've tried this:

&SURF ID='top_wall', RAMP_T='top_temp_ramp' /
&RAMP ID='top_temp_ramp', T=0.0, F=50.0 /
&RAMP ID='top_temp_ramp', T=0.00067, F=47.15 /
&RAMP ID='top_temp_ramp', T=0.0133, F=45.53 /
&RAMP ID='top_temp_ramp', T=0.0533, F=43.92 /
&RAMP ID='top_temp_ramp', T=0.20, F=41.59 /
&RAMP ID='top_temp_ramp', T=0.40, F=39.06 /
&RAMP ID='top_temp_ramp', T=0.60, F=36.67 /
&RAMP ID='top_temp_ramp', T=0.80, F=34.17 /
&RAMP ID='top_temp_ramp', T=0.90, F=32.45 /
&RAMP ID='top_temp_ramp', T=0.95, F=31.04 /
&RAMP ID='top_temp_ramp', T=0.99, F=29.71 /
&RAMP ID='top_temp_ramp', T=1.0, F=25.0 /

&SURF ID='bottom_wall', RAMP_T='bottom_temp_ramp' /
&RAMP ID='bottom_temp_ramp', T=0.0, F=50.0 /
&RAMP ID='bottom_temp_ramp', T=0.00067, F=46.91 /
&RAMP ID='bottom_temp_ramp', T=0.0133, F=45.15 /
&RAMP ID='bottom_temp_ramp', T=0.0533, F=41.82 /
&RAMP ID='bottom_temp_ramp', T=0.20, F=36.42 /
&RAMP ID='bottom_temp_ramp', T=0.40, F=30.92 /
&RAMP ID='bottom_temp_ramp', T=0.60, F=27.42 /
&RAMP ID='bottom_temp_ramp', T=0.80, F=24.09 /
&RAMP ID='bottom_temp_ramp', T=0.90, F=22.45 /
&RAMP ID='bottom_temp_ramp', T=0.95, F=21.26 /
&RAMP ID='bottom_temp_ramp', T=0.99, F=20.09 /
&RAMP ID='bottom_temp_ramp', T=1.0, F=20.0 /

&OBST XB=0.0,0.75,0.75,0.75,0.0,0.01, SURF_ID='top_wall' /
&OBST XB=0.0,0.75,0.0,0.0,0.0,0.01, SURF_ID='bottom_wall' /

**Comment by drjfloyd** [comment]
Please read the User's Guide. There are examples for doing just this.

**Reply by appheadspace** [reply]
Thanks for your reply! I've tried different approaches after seeing user guide only, but still not able to get temp distribution. Kindly help me.

**Comment by mcgratta** [comment]
This is not how a temperature ramp works.
```
&SURF ID='...', TMP_FRONT

**Reply by appheadspace** [reply]
Thank you for your reply!

**Approach - 1:::**

&OBST XB=0,0.750,-0.00005,0.00005,0,0 ,  SURF_ID='bottom_wall' /
&SURF ID = 'bottom_wall', COLOR = 'GRAY'/


&OBST XB =0,0,-0.00005,0.00005,0,0 ,SURF_ID='BW1'/  
&SURF ID = 'BW1',TMP_FRONT =50/

&OBST XB =0.00015,0.00015,-0.00005,0.00005,0,0 ,SURF_ID='BW2'/  
&SURF ID = 'BW2',TMP_FRONT =47.96/

&OBST XB =0.005025,0.005025,-0.00005,0.00005,0,0 ,SURF_ID='BW3'/  
&SURF ID = 'BW3',TMP_FRONT =47.33/


**Approach-2:::**

&OBST XB=0,0.750,-0.00005,0.00005,0,0 ,  SURF_ID='bottom_wall' /
&SURF ID = 'bottom_wall',  COLOR = 'GRAY', RAMP_T = Bottom_Temp_Ramp, TMP_FRONT = 30/

&RAMP ID='Bottom_Temp_Ramp',  T=0, F=1.6667
&RAMP ID='Bottom_Temp_Ramp',  T=0.00015, F=1.5987
&RAMP ID='Bottom_Temp_Ramp',  T=0.005025, F=1.5777


I've tried both above approaches, but bottom wall temperatures are showing 30 at all locations. (PS: In my case INIT temp and TMPA , both are 30)

Kindly guide me.

**Reply by drjfloyd** [reply]
<img width="428" alt="image" src="https://github.com/user-attachments/assets/92bf33b5-3dc5-4709-b8c2-cf92ab78c95d">
